jetbrains-rider/src/software/aws/toolkits/jetbrains/services/lambda/dotnet/DotNetSamDebugSupport.kt

```diff
@@ -24,60 +31,47 @@ import com.jetbrains.rd.util.put
 import com.jetbrains.rd.util.reactive.adviseUntil
 import com.jetbrains.rdclient.protocol.RdDispatcher
 import com.jetbrains.rider.debugger.RiderDebuggerWorkerModelManager
-import com.jetbrains.rider.model.debuggerWorker.DotNetCoreExeStartInfo
-import com.jetbrains.rider.model.debuggerWorker.DotNetCoreInfo
+import com.jetbrains.rider.model.debuggerWorker.DotNetCoreAttachStartInfo
 import com.jetbrains.rider.model.debuggerWorker.DotNetDebuggerSessionModel
 import com.jetbrains.rider.model.debuggerWorkerConnectionHelperModel
 import com.jetbrains.rider.projectView.solution
 import com.jetbrains.rider.run.IDebuggerOutputListener
 import com.jetbrains.rider.run.bindToSettings
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.withContext
 import org.jetbrains.concurrency.AsyncPromise
 import org.jetbrains.concurrency.Promise
 import software.amazon.awssdk.services.lambda.model.Runtime
-import software.aws.toolkits.core.utils.error
-import software.aws.toolkits.core.utils.getLogger
-import software.aws.toolkits.core.utils.trace
 import software.aws.toolkits.jetbrains.services.lambda.execution.local.SamDebugSupport
 import software.aws.toolkits.jetbrains.services.lambda.execution.local.SamRunningState
+import software.aws.toolkits.jetbrains.utils.ApplicationThreadPoolScope
 import software.aws.toolkits.jetbrains.utils.DotNetDebuggerUtils
+import software.aws.toolkits.jetbrains.utils.getCoroutineUiContext
+import software.aws.toolkits.jetbrains.utils.spinUntilResult
 import software.aws.toolkits.resources.message
-import java.io.File
-import java.io.OutputStream
 import java.net.InetAddress
+import java.time.Duration
 import java.util.Timer
 import kotlin.concurrent.schedule
 
 /**
  * Rider uses it's own DebuggerWorker process that run under .NET with extra parameters to configure:
- *    - path to runtime
- *    - path to DLL
  *    - debugger mode (client/server)
  *    - frontend port
  *    - backend port
  *
- * Debugger is launched via SAM CLI command: '$ sam local invoke' and pass debugger parameters --debugger-path and --debugger-args.
- * Under SAM CLI this command is translated into:
- *    - volume that is mounted in Docker to launch debugger (--debugger-path).
- *      This volume is set to path with debugger process used in Rider - DebuggerWorker - inside Rider SDK.
- *    - additional debugger arguments that are applied when launching debugger inside Docker (--debugger-args):
- *      '$ dotnet <debug_args_list> MockBootstraps.dll --debugger-spin-wait'
- *
- * We use <debugger_args_list> to inject a simple custom dotnet core application 'JetBrains.Rider.Debugger.Launcher'
- * that run Rider's debugger. This app starts Rider's debugger under Mono runtime that is bundled with Rider SDK (runtime.sh).
```

Comment:
what's the change that removes the need to run under the bundled mono runtime?

Reply:
Starting in 193, the debugger can run in dotnet core. We use it for cloud debug in that mode too